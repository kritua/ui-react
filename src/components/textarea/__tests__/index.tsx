import React from 'react';
import { create } from 'react-test-renderer';

import { InputTextarea } from '../';

it('Default', () => {
    const tree = create(
        <InputTextarea
            name="default"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('Required', () => {
    const tree = create(
        <InputTextarea
            name="default"
            required={true}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('Label', () => {
    const tree = create(
        <InputTextarea
            name="default"
            label="Input label"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('Error & Message string', () => {
    const tree = create(
        <InputTextarea
            name="default"
            label="Input label"
            isError={true}
            elError="Message"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('Error & Message ReactNode', () => {
    const tree = create(
        <InputTextarea
            name="default"
            label="Input label"
            isError={true}
            elError={(
                <div>Custom error</div>
            )}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('Description string', () => {
    const tree = create(
        <InputTextarea
            name="default"
            elDescription="Description string"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('Description ReactNode', () => {
    const tree = create(
        <InputTextarea
            name="default"
            elDescription={(
                <div>Description ReactNode</div>
            )}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('disabled', () => {
    const tree = create(
        <InputTextarea
            name="default"
            disabled={true}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('disabled & readOnly', () => {
    const tree = create(
        <InputTextarea
            name="default"
            disabled={true}
            readOnly={true}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

it('readOnly', () => {
    const tree = create(
        <InputTextarea
            name="default"
            readOnly={true}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});
