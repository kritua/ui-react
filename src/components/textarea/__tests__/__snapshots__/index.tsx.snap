// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default 1`] = `
<label
  className="input-textarea"
>
  <textarea
    className="input-textarea__input"
    name="default"
    tabIndex={0}
  />
</label>
`;

exports[`Description ReactNode 1`] = `
<label
  className="input-textarea"
>
  <textarea
    className="input-textarea__input"
    name="default"
    tabIndex={0}
  />
  <div>
    Description ReactNode
  </div>
</label>
`;

exports[`Description string 1`] = `
<label
  className="input-textarea"
>
  <textarea
    className="input-textarea__input"
    name="default"
    tabIndex={0}
  />
  <span
    className="input-textarea__description"
  >
    Description string
  </span>
</label>
`;

exports[`Error & Message ReactNode 1`] = `
<label
  className="input-textarea"
>
  <div
    className="input-textarea__label"
  >
    Input label
  </div>
  <textarea
    className="input-textarea__input input-textarea__input_is-error"
    name="default"
    tabIndex={0}
  />
  <div>
    Custom error
  </div>
</label>
`;

exports[`Error & Message string 1`] = `
<label
  className="input-textarea"
>
  <div
    className="input-textarea__label"
  >
    Input label
  </div>
  <textarea
    className="input-textarea__input input-textarea__input_is-error"
    name="default"
    tabIndex={0}
  />
  <span
    className="input-textarea__error"
  >
    Message
  </span>
</label>
`;

exports[`Label 1`] = `
<label
  className="input-textarea"
>
  <div
    className="input-textarea__label"
  >
    Input label
  </div>
  <textarea
    className="input-textarea__input"
    name="default"
    tabIndex={0}
  />
</label>
`;

exports[`Required 1`] = `
<label
  className="input-textarea"
>
  <textarea
    className="input-textarea__input"
    name="default"
    required={true}
    tabIndex={0}
  />
</label>
`;

exports[`disabled & readOnly 1`] = `
<label
  className="input-textarea"
>
  <textarea
    className="input-textarea__input"
    disabled={true}
    name="default"
    readOnly={true}
    tabIndex={-1}
  />
</label>
`;

exports[`disabled 1`] = `
<label
  className="input-textarea"
>
  <textarea
    className="input-textarea__input"
    disabled={true}
    name="default"
    tabIndex={-1}
  />
</label>
`;

exports[`readOnly 1`] = `
<label
  className="input-textarea"
>
  <textarea
    className="input-textarea__input"
    disabled={true}
    name="default"
    readOnly={true}
    tabIndex={-1}
  />
</label>
`;
